utils/vptr/include/virtual_ptr.hpp

```diff
@@ -267,7 +267,12 @@ class PointerMapper {
   /**
    * Constructs the PointerMapper structure.
    */
-  PointerMapper() : m_pointerMap{}, m_freeList{} {};
+  PointerMapper(base_ptr_t baseAddress = 4096)
+      : m_pointerMap{}, m_freeList{}, m_baseAddress{baseAddress} {
+    if (m_baseAddress == 0) {
+      throw std::out_of_range(std::string("Base address cannot be zero"));
```

Comment:
You're too right! I actually went onto cppreference, looked at the exceptions, thought to myself "this is invalid arg!" and wrote out of range because I'm a damn fool. Thanks!